proxy/tests/discovery.rs

```diff
@@ -201,20 +199,13 @@ fn outbound_updates_newer_services() {
     // from the controller
     let client1 = client::http2(proxy.outbound, "disco.test.svc.cluster.local");
 
-    // This HTTP2 client tries to talk to our HTTP1 server, and the server
-    // will return an error (see above TODO).
-    //
-    // The reason to do this request at all is because the proxy will create
-    // an H2 service mapping when it sees an H2 request, and we want to test
-    // that when it sees H1 and tries to create a new mapping, the existing
-    // known Discovery information is shared with it.
-    let res = client1.request(&mut client1.request_builder("/h1"));
-    assert_eq!(res.status(), 500);
+    // Depending on the version of `hyper` we're using, protocol upgrades may or
+    // may not be supported yet, so this may have a response status of either 200
+    // or 500. Ignore the status code for now, and just expect that making the
+    // request doesn't *error* (which `client.request` does for us).
+    let _res = client1.request(&mut client1.request_builder("/h1"));
+    // assert_eq!(res.status(), 200);
 
-
-    // a new HTTP1 service needs to be build now, while the HTTP2
-    // service already exists, so make sure previously sent addrs
-    // get into the newer service
```

Comment:
This comment is still relevant, as it explains why the test is here.

Reply:
Whoops! I'll put it back, thanks.